feat(apple): SwiftUI documentation (#6046)
getsentry/sentry-docs: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx

---
title: SwiftUI Instrumentation
sidebar_order: 12
description: "Learn how to monitor the performance of your SwiftUI views."
---

<Note>

>     This is an experimental feature.

Author: ```suggestion
    <Include name="beta-note.mdx" />
```

> </Note>

Reviewer: Please consider replacing this with `<Include name="alpha-note.mdx" />` or `<Include name="beta-note.mdx" />` or add an extra reusable include like the previous ones.

> <Note>
>     This is an experimental feature.
> </Note>

Author: ```suggestion
<Include name="beta-note.mdx" />
```

You can monitor the performance of your views in a SwiftUI project with SentrySwiftUI library.

> Install it with CocoaPod:

Reviewer: `m`: I think we should point out that SwiftUI needs an extra package. If I remember correctly, SwiftUI also includes Sentry, so we should clarify that we don't need both.

Maybe it is better to add this info to each of the pages in https://docs.sentry.io/platforms/apple/guides/ios/install/. So basically the code samples there would mention that there is also a package for SwiftUI.

```
platform :ios, '9.0'
use_frameworks! # This is important

target 'YourApp' do
  pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => '7.31.5'
  # If you want SwiftUI or something like that
  pod 'SentrySwiftUI', :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => '7.31.5'
end
```

```ruby
platform :ios, '13.0'
use_frameworks! # This is important

target 'YourApp' do
  pod 'SentrySwiftUI'
end
```

Or by importing sentry-cocoa repository (https://github.com/getsentry/sentry-cocoa) with SPM.

> In order to use it, you need to wrap your view with `SentryTracerView`, like this:

Reviewer: ```suggestion
In order to use it, you need to wrap your view with `SentryTracedView`, like this:
```

```swift {tabTitle:Swift}

> import SentrySwiftUI

Reviewer: `m`: I think one more example with nested views would be great.

var body: some View {

>     SentryTracedView("<NAME YOU TRANSACTION>"){

Reviewer: `m`: It's not only the name of the transaction, it's also the name of a span if you nest them.
```suggestion
    SentryTracedView("<NAME YOUR TRANSACTION>"){
```

>         //... Your view content

Reviewer: `l`: I would add an actual view

    }
}
```

> You may omit the transaction name, Sentry will use the root view type as name.

Reviewer: `m`: Please add an example for a Button or a Text.

> Alternatively you can use a modifier like this:

Reviewer: ```suggestion
Alternatively, you can use a modifier like this:
```

> Alternatively you can use a modifier like this:

Reviewer: `m`: We have to point out that this does the same as `SentryTracedView{ ... }`

Author: Well, that's what `Alternatively` means right?

```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {
    VStack {

    }.sentryTrace("<NAME YOUR TRANSACTION>")
}

> ```

Reviewer: `m`: I think we should add a screenshot of a sample transaction with nested views.

sidebar_order: 12
description: "Learn how to monitor the performance of your SwiftUI views."
---

<Include name="beta-note.mdx" />

You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink> or CocoaPods:

```ruby
platform :ios, '13.0'
use_frameworks! # This is important

target 'YourApp' do
  pod 'SentrySwiftUI'
end
```

> If you plan to add _SentrySwiftUI_, you don`t need to also add _Sentry_ library since the former dependents on the latter.

In order to start monitoring the performance of your views, you need to wrap it with `SentryTracedView`, like this:
```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {
    SentryTracedView("<Name of your view>"){
        List {
            // Your SwiftUI code ...
        }
    }
}
```

You may omit the transaction name, Sentry will use the root view type as name, in the previous example the transaction name would be `List`.

Alternatively, you can use a modifier like this:

```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {
    List {

> 

Author: ```suggestion
// Your SwiftUI code ...
```

    }.sentryTrace("<NAME YOUR TRANSACTION>")
}
```

It is possible to use nested `SentryTracedView` to have a more granular transaction.

```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {
    SentryTracedView("My Awesome Screen"){
        HStack {
            SentryTracedView("Menu") {

>                 List{
>                 }
>             }
>             SentryTracedView("Content") {
>                 VStack{
>                 }

Author: ```suggestion
                List{
                // Your SwiftUI code ...
                }
            }
            SentryTracedView("Content") {
                VStack{
                // Your SwiftUI code ...
                }
```

---
title: SwiftUI Instrumentation
sidebar_order: 12
description: "Learn how to monitor the performance of your SwiftUI views."
---

<Include name="beta-note.mdx" />

You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink> or CocoaPods:

```ruby
platform :ios, '13.0'
use_frameworks! # This is important

target 'YourApp' do
  pod 'SentrySwiftUI'
end

> ```

Reviewer: `h`: As long as we don't fix https://github.com/getsentry/sentry-cocoa/issues/2581, we should only mention SPM here, please.

Author: We need to fix it before merging this documents

> You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink> or CocoaPods:
> 
> ```ruby
> platform :ios, '13.0'
> use_frameworks! # This is important
> 
> target 'YourApp' do
>   pod 'SentrySwiftUI'
> end
> ```

Author: ```suggestion
You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink>.
```

> If you plan to add _SentrySwiftUI_, you don`t need to also add _Sentry_ library since the former dependents on the latter.

In order to start monitoring the performance of your views, you need to wrap it with `SentryTracedView`, like this:
```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {
    SentryTracedView("<Name of your view>"){
        List {
            // Your SwiftUI code ...
        }
    }
}
```

You may omit the transaction name, Sentry will use the root view type as name, in the previous example the transaction name would be `List`.

Alternatively, you can use a modifier like this:

```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {
    List {

> // Your SwiftUI code ...
>     }.sentryTrace("<NAME YOUR TRANSACTION>")

Reviewer: `m`:
```suggestion
        // Your SwiftUI code ...
    }.sentryTrace("<Name of your view>")
```

---
title: SwiftUI Instrumentation
sidebar_order: 12
description: "Learn how to monitor the performance of your SwiftUI views."
---

<Include name="beta-note.mdx" />

> You can monitor the performance of your views in a SwiftUI project with SentrySwiftUI library.
> 
> Install it with CocoaPod:

Reviewer: If you apply this suggestion you can ditch `Or by importing sentry-cocoa repository (https://github.com/getsentry/sentry-cocoa) with SPM.` below
```suggestion
You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink> or CocoaPods:
```

```ruby
platform :ios, '13.0'
use_frameworks! # This is important

target 'YourApp' do
  pod 'SentrySwiftUI'
end
```

> Or by importing sentry-cocoa repository (https://github.com/getsentry/sentry-cocoa) with SPM.
> 

Author: ```suggestion
```

> If you plan to add _SentrySwiftUI_, you don`t need to also add _Sentry_ library since the former dependents on the latter.

In order to start monitoring the performance of your views, you need to wrap it with `SentryTracedView`, like this:
```swift {tabTitle:Swift}

import SentrySwiftUI

var body: some View {

>     SentryTracedView("<NAME YOUR TRANSACTION>"){

Reviewer: `m`: I think we could simply change this to pass in your view name, so users don't have to think about transaction names. They simply put in their view names, and Sentry does the rest. I think we should add a sentence, that the SDK will use the root view name as transaction name. We then would only need one code sample with nested views. Not two.
```suggestion
    SentryTracedView("<Name of your view>"){
```

I think we should also rename the parameter `transactionName` of `SentryTracedView`. I created an issue for that https://github.com/getsentry/sentry-cocoa/issues/2608.

>         List {
>         }

Reviewer: `m`: I would add a comment as a placeholder for each code sample.
```suggestion
        List {
            // Your SwiftUI code ...
        }
```
